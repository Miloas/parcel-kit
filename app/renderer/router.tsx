import * as React from 'react'
import { HashRouter, Route, Switch } from 'react-router-dom'

import { Hello, World } from './pages'

export const App = () => {
  return (
    <HashRouter hashType='noslash'>
      <Switch>
        <Route exact={ true } path='/' component={ Hello } />
        <Route exact={ true } path='/next' component={ World } />
        <Route component={ () => <h1>204 No Content</h1> } />
      </Switch>
    </HashRouter>
  )
}
