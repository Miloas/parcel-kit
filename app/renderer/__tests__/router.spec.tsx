import * as React from 'react'
import * as Renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import { App } from '../router'

test('route /', () => {
  const component = Renderer.create(
    <MemoryRouter initialEntries={ ['/'] }><App /></MemoryRouter>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('route /next', () => {
  const component = Renderer.create(
    <MemoryRouter initialEntries={ ['/next'] }><App /></MemoryRouter>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
