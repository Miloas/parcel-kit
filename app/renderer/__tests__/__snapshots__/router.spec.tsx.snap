// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`route / 1`] = `
<div
  style={
    Object {
      "marginTop": 100,
      "textAlign": "center",
    }
  }
>
  <h1>
    Hello
  </h1>
  <h3>
    <a
      href="#next"
      onClick={[Function]}
    >
      next page
    </a>
  </h3>
</div>
`;

exports[`route /next 1`] = `
<div
  style={
    Object {
      "marginTop": 100,
      "textAlign": "center",
    }
  }
>
  <h1>
    Hello
  </h1>
  <h3>
    <a
      href="#next"
      onClick={[Function]}
    >
      next page
    </a>
  </h3>
</div>
`;
